internal/vsp/vsp.go

```diff
@@ -238,39 +239,69 @@ func (c *Client) Process(ctx context.Context, ticketHash *chainhash.Hash, feeTx
 // be specified, instead of using the client's default policy.
 func (c *Client) ProcessWithPolicy(ctx context.Context, ticketHash *chainhash.Hash, feeTx *wire.MsgTx,
 	policy Policy) error {
-	fp := c.feePayment(ticketHash, policy)
-	if fp == nil {
-		err := c.Wallet.UpdateVspTicketFeeToErrored(ctx, ticketHash, c.client.url, c.client.pub)
-		if err != nil {
-			return err
-		}
-		return fmt.Errorf("fee payment cannot be processed")
-	}
-	fp.mu.Lock()
-	if fp.feeTx == nil {
-		fp.feeTx = feeTx
+
+	vspTicket, err := c.Wallet.VSPTicketInfo(ctx, ticketHash)
+	if err != nil && !errors.Is(err, errors.NotExist) {
+		log.Error(err)
+		return nil
 	}
-	fp.mu.Unlock()
-	err := fp.receiveFeeAddress()
-	if err != nil {
-		err := c.Wallet.UpdateVspTicketFeeToErrored(ctx, ticketHash, c.client.url, c.client.pub)
+
+	switch udb.FeeStatus(vspTicket.FeeTxStatus) {
```

Comment:
This needs to also check first whether `vspTicket` is `!= nil`, otherwise this may panic if `ProcessWithPolicy` is ever somehow called with a tx hash that is not a previously added ticket. 

Either that, or change the previous if to _not_ ignore `errors.NotExist`.